src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,532 @@
+
+    /**
+     * Tag to store the fields stage.
+     */
+    private static final String TAG_STAGE = "stage";
+
+    /**
+     * Amount of rows in the player inventory.
+     */
+    private static final int PLAYER_INVENTORY_ROWS = 3;
+
+    /**
+     * Amount of columns in the player inventory.
+     */
+    private static final int PLAYER_INVENTORY_COLUMNS = 9;
+
+    /**
+     * Initial x-offset of the inventory slot.
+     */
+    private static final int PLAYER_INVENTORY_INITIAL_X_OFFSET = 8;
+
+    /**
+     * Initial y-offset of the inventory slot.
+     */
+    private static final int PLAYER_INVENTORY_INITIAL_Y_OFFSET = 84;
+
+    /**
+     * Each offset of the inventory slots.
+     */
+    private static final int PLAYER_INVENTORY_OFFSET_EACH = 18;
+
+    /**
+     * Initial y-offset of the inventory slots in the hotbar.
+     */
+    private static final int PLAYER_INVENTORY_HOTBAR_OFFSET = 142;
+
+    /**
+     * The max width/length of a field.
+     */
+    private static final int MAX_RANGE = 5;
+    /**
+     * The colony of the field.
+     */
+    private final Colony   colony;
+    /**
+     * The fields location.
+     */
+    private       BlockPos location;
+    /**
+     * Has the field be taken by any worker?
+     */
+    private boolean taken = false;
+
+    /**
+     * Checks if the field needsWork (Hoeig, Seedings, Farming etc).
+     */
+    private boolean needsWork = false;
+
+    /**
+     * Has the field been planted?
+     */
+    private FieldStage fieldStage = FieldStage.EMPTY;
+
+    /**
+     * The length to plus x of the field.
+     */
+    private int lengthPlusX;
+
+    /**
+     * The width to plus z of the seed.
+     */
+    private int widthPlusZ;
+
+    /**
+     * The length to minus xof the field.
+     */
+    private int lengthMinusX;
+
+    /**
+     * The width to minus z of the seed.
+     */
+    private int widthMinusZ;
+
+    /**
+     * The inventorySlot of the field.
+     */
+    private InventoryField inventory;
+
+    /**
+     * Private constructor to create field from NBT.
+     *
+     * @param colony the colony the field belongs to.
+     */
+    private Field(Colony colony)
+    {
+        super();
+        this.colony = colony;
+    }
+
+    /**
+     * Creates an instance of our field container, this may be serve to open the GUI.
+     *
+     * @param scarecrowTileEntity       the tileEntity of the field containing the inventory.
+     * @param playerInventory the player inventory.
+     * @param world           the world.
+     * @param location        the position of the field.
+     */
+    public Field(ScarecrowTileEntity scarecrowTileEntity, InventoryPlayer playerInventory, World world, BlockPos location)
+    {
+        super();
+        this.colony = ColonyManager.getColony(world, location);
+        this.location = location;
+        this.inventory = scarecrowTileEntity.getInventoryField();
+
+        addSlotToContainer(new Slot(inventory, 0, X_OFFSET, Y_OFFSET));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this call from a constructor to the overridable "addSlotToContainer" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1699)